### ISSUE TITLE
npm run android-simple.js does not work while mocha android-simple.js works

### ISSUE THREAD

**Issue by rahulmr** [appium/sample-code#95]
Hello, 
I am getting issue with running the examples for nodejs.
`npm run android-simple.js` does not work while `mocha android-simple.js `works fine.

**Not working** 
```
rahul.raut1@mywindows MINGW64 /d/projects/automation/sample-code/sample-code/examples/node (master)
$ npm run android-simple.js
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "D:\\nodejs\\node.exe" "D:\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "android-simple.js"
npm ERR! node v6.2.2
npm ERR! npm  v3.10.5

npm ERR! missing script: android-simple.js
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     D:\projects\automation\sample-code\sample-code\examples\node\npm-debug.log

rahul.raut1@mywindows MINGW64 /d/projects/automation/sample-code/sample-code/examples/node (master)

```

**Working** 

```
rahul.raut1@mywindows MINGW64 /d/projects/automation/sample-code/sample-code/examples/node (master)
$ mocha android-simple.js


  android simple
 > CALL init({"browserName":"","appium-version":"1.5.3","platformName":"Android","platformVersion":"5.1","deviceName":"160530023779","app":"http://appium.github.io/appium/assets/ApiDemos-debug.apk"})
 > POST /session {"desiredCapabilities":{"browserName":"","appium-version":"1.5.3","platformName":"Android","platformVersion":"5.1","deviceName":"160530023779","app":"http://appium.github.io/appium/assets/ApiDemos-debug.apk"}}

Driving the web on session: deba8719-5414-4fad-89a4-62cdcc2125a8

 > RESPONSE init({"browserName":"","appium-version":"1.5.3","platformName":"Android","platformVersion":"5.1","deviceName":"160530023779","app":"http://appium.github.io/appium/assets/ApiDemos-debug.apk"}) "deba8719-5414-4fad-89a4-62cdcc2125a8",{"platform":"LINUX","webStorageEnabled":false,"takesScreenshot":true,"javascriptEnabled":true,"databaseEnabled":false,"networkConnectionEnabled":true,"locationContextEnabled":false,"warnings":{},"desired":{"browserName":"","appium-version":"1.5.3","platformName":"Android","platformVersion":"5.1","deviceName":"160530023779","app":"http://appium.github.io/appium/assets/ApiDemos-debug.apk"},"browserName":"","appium-version":"1.5.3","platformName":"Android","platformVersion":"5.1","deviceName":"160530023779","app":"http://appium.github.io/appium/assets/ApiDemos-debug.apk","deviceUDID":"160530023779","appPackage":"io.appium.android.apis","appWaitPackage":"io.appium.android.apis","appActivity":"io.appium.android.apis.ApiDemos","appWaitActivity":"io.appium.android.apis.ApiDemos"}
 > CALL setImplicitWaitTimeout(3000)
 > POST /session/:sessionID/timeouts/implicit_wait {"ms":3000}
 > RESPONSE setImplicitWaitTimeout(3000)
 > CALL elementByAccessibilityId("Graphics")
 > POST /session/:sessionID/element {"using":"accessibility id","value":"Graphics"}
    √ should find an element (80ms)
 > CALL quit()
 > DELETE /session/:sessionID

Ending your web drivage..

 > RESPONSE quit()


  1 passing (24s)


rahul.raut1@mywindows MINGW64 /d/projects/automation/sample-code/sample-code/examples/node (master)

```

Is it possible to improve the documentation or is it something I am missing out?

Thanks for help and support.

**Comment by imurchie** [comment]
Where are you running the npm command? It has to be run in the directory with the node sample code. Mocha will search sub-directories, but the npm command will not.

**Comment by rahulmr** [comment]
I have attached the traces, I am running the `npm run` command from `sample-code/node` folder. I am using Git Bash on Windows 7.

**Comment by wmhilton** [comment]
**TLDR:** It's `npm run android-simple` not `npm run android-simple.js`

Hi @rahulmr. I don't work on appium but I saw your question and I think I can answer it. The argument to "npm run" is NOT a file name, but actually an arbitrary name listed in package.json. So "npm run foo" causes npm to look in package.json for a script labeled "foo". See...

```
$ cd sample-code/examples/node
$ cat package.json
{
  "name": "appium-sample-code",
  "version": "1.0.0",
  "description": "Appium Sample Code",
  "main": "index.js",
...
  "scripts": {
    "ios-simple": "mocha ios-simple.js",
    "ios-complex": "mocha ios-complex.js",
    "ios-webview": "mocha ios-webview.js",
    "ios-actions": "mocha ios-actions.js",
    "ios-local-server": "mocha ios-local-server.js",
    "ios-selenium-webdriver-bridge": "mocha ios-selenium-webdriver-bridge.js",
    "ios-yiewd": "mocha --harmony ios-yiewd.js",
    "android-simple": "mocha android-simple.js",             <-------- HERE IT IS
    "android-complex": "mocha android-complex.js",
    "android-webview": "mocha android-webview.js",
    "android-actions": "mocha android-actions.js",
    "selendroid-simple": "mocha selendroid-simple.js"
  }
}
```

This project is a little unusual, typically the scripts look more like this:
```
"scripts": {
  "start": "node lib/server.js",
  "test": "mocha tests -R",
  "build":  "babel --presets es2015,stage-0 -d lib/ src/",
  "dev": "node-dev",
  "prepublish": "npm run build && npm run test"
}
```
You can see how in the typical case, these names are actually helpful. For instance, everyone knows "npm run start" should start the project, and "npm run test" is what testing frameworks like Travis-CI will run. Certain script names like "prepublish" or "postinstall" are triggered at times by npm itself to perform certain operations. See the complete list here: https://docs.npmjs.com/misc/scripts

**Comment by rahulmr** [comment]
I am grateful. Thanks a lot! Little mistakes can ruin things :)
